app/mobile-content/exhibits/exhibits.component.ts

```diff
@@ -16,73 +22,124 @@ export class ExhibitsComponent implements OnInit {
   exhibits: Exhibit[];
   routes: string[];
   statuses = Status.getValuesForSearch();
+  private translatedResponse: string;
+  private exhibitCache = new Map<number, Exhibit[]>();
+
 
   // search parameters
   searchQuery = '';
-  selectedRoute = 'ALL';
-  selectedStatus: 'ALL' | statusType = 'ALL';
+  selectedStatus = 'ALL';
   showingSearchResults = false;
 
   // pagination parameters
+  exhibitsPerPage = 10;
   currentPage = 1;
-  pageSize = 10;
-  totalItems: number;   // must be fetched from server
+  totalItems: number;
 
   // dialogs
   private createDialogRef: MdDialogRef<CreateExhibitDialogComponent>;
   private deleteDialogRef: MdDialogRef<DeleteExhibitDialogComponent>;
 
-  constructor(private dialog: MdDialog) {}
+  constructor(private dialog: MdDialog,
+    private exhibitService: ExhibitService,
+    public  router: Router,
+    private toasterService: ToasterService,
+    private translateService: TranslateService) {}
 
-  ngOnInit() {
-    // TODO: replace dummy data with appropriate API calls
-    this.exhibits = new Array(30);
-    this.totalItems = this.exhibits.length;
-    for (let i = 0; i < this.exhibits.length; i++) {
-      this.exhibits[i] = Exhibit.getRandom();
+    ngOnInit() {
+      this.getPage(1);
     }
-    this.routes = ['ALL', 'Route 1', 'Route 2', 'Route 3', 'Route 4'];
```

Comment:
Please leave at least the default option (all routes) here. The list of routes can be added after the route service is merged.